syntax = "proto3";
import "db_models.proto";



message ClientValidationList{
    map<int64, bool> Clients = 1;
}


message Payment{
    int64 ClientID = 1;
    int64 TransactionID = 2; 
    int64 TransportID = 3;
}


message CompletedPayment{
    string price = 1;
    TransportType type = 2;
    string title = 3; 
    int32 date = 4;
    int64 id = 5;
    Status status = 6; 
}

message RecentPaymentsRequest {
    int64 client_id = 1; 
}

message RecentPaymentsResponce{
    repeated CompletedPayment Payments = 1;
}

message UserInfoRequest {
    int64 client_id = 1; 
}

message UserInfoResponse {
    enum PayStatus {
        Available = 0;
        Blocked = 1;
    }
    PayStatus status = 1;
    string balance = 2;
}

message SessionOpenRequest{
    int32 driver_id = 1;
    int32 transport_id = 2;
    int32 trace_id = 3;
}


message SessionOpenResponce{
    int32 session_id = 1;
}


message SessionCloseRequest{
    int32 session_id = 1;
}


message GetPriceRequest{
    int32 transport_id = 1;
}


message GetPriceResponse{
    string Price = 1;
    int32 session_id = 2;
}


message TransactionSyncRequest{
    repeated Payment Payments = 1;
}

message RefilRequest{
    int64 client_id = 1 ;
    int32 value = 2;
}


message TrafficStatPoint{
    uint32 start_time = 1;
    uint32 end_time = 2;
    int32 count = 3;
}


message TrafficStatsResponce{
    int32 trace_id = 1;
    repeated TrafficStatPoint Stats = 2;
}

message TrafficStatsRequest{
    int32 trace_id = 1; 
}
